lib/jekyll/watcher.rb

```diff
@@ -18,12 +18,13 @@ module Watcher
     # Returns nothing.
     def watch(options, site = nil)
       ENV["LISTEN_GEM_DEBUGGING"] ||= "1" if options['verbose']
+      @source = options['source']
```

Comment:
this needs an `attr_reader` defined for it and then that reader method needs to be used everywhere. IOW, define the attr_reader like so:
```ruby
attr_reader :source
```

and then replace all the `@source` with `source` and it should be good.